### DISCUSSION TITLE
check deprecated function

### DISCUSSION THREAD

**Discussion by Nina-KNN** [detekt/detekt#5497] [Q&A] [answered]
Can I get a option of function(like deprecated), if i have a name or call expression?

I need to create a rule that will be check and show error if developer call deprecated function in his class. It should be work for finding all deprecated function for full project (in library too) via detekt check.

**Comment by BraisGabin** [comment, marked as answer]
I don't get exactly what you are asking for. What are you missing from the standard [`Deprecation` rule](https://detekt.dev/docs/next/rules/potential-bugs#deprecation)?

**Reply by Nina-KNN** [reply]
Sorry for the long answer. 
Thank you so much for your help. Yes, you are right, this rule should be doing what I need. But it doesn't find any deprecated methods in code.
I made this rule active in configuration file.

For examle: 
- resources.**getColor(R.color.red)** - is Deprecated
Android Studio highlights this method as deprecated, but this error is not shown in the Detekt's report

**Reply by cortinico** [reply]
This rule requires type resolution: https://detekt.dev/docs/gettingstarted/type-resolution

**Reply by Nina-KNN** [reply]
yo) yes, now it works perfect) Thank you so much for your help)
